tools/ossf_best_practices/passing_criteria.md

```diff
@@ -0,0 +1,397 @@
+
+**Met**
+
+
+> A test suite SHOULD be invocable in a standard way for that language.
+
+**Met**
+make or batch file, executed using python
+
+
+> It is SUGGESTED that the test suite cover most (or ideally all) the code branches, input fields, and functionality.
+
+**Met**
+
+
+> It is SUGGESTED that the project implement continuous integration (where new or changed code is frequently integrated into a central code repository and automated tests are run on the result).
+
+**Met**
+https://ci.nodejs.org/
+
+
+## New functionality testing
+
+> The project MUST have a general policy (formal or not) that as major new functionality is added to the software produced by the project, tests of that functionality should be added to an automated test suite.
+
+**Met** 
+https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-5-test
+
+> The project MUST have evidence that the test_policy for adding tests has been adhered to in the most recent major changes to the software produced by the project.
+
+**Met**
+
+> It is SUGGESTED that this policy on adding tests (see test_policy) be documented in the instructions for change proposals
+
+**Met**
+
+## Warning flags
+
+> The project MUST enable one or more compiler warning flags, a "safe" language mode, or use a separate "linter" tool to look for code quality errors or common simple mistakes, if there is at least one FLOSS tool that can implement this criterion in the selected language.
+
+**Met**
+-Wall for compiling, c++ and JS linting as part of test suite
+
+
+> The project MUST address warnings.
+
+**Met**
+
+> It is SUGGESTED that projects be maximally strict with warnings in the software produced by the project, where practical.
+
+**Met**
+
+# Security
+
+## Secure development knowledge
+
+> The project MUST have at least one primary developer who knows how to design secure software. (See ‘details’ for the exact requirements.)
+
+**Met**
+
+> At least one of the project's primary developers MUST know of common kinds of errors that lead to vulnerabilities in this kind of software, as well as at least one method to counter or mitigate each of them.
+
+**Met**
+
+## Use basic good cryptographic practices
+
+
+> The software produced by the project MUST use, by default, only cryptographic protocols and algorithms that are publicly published and reviewed by experts (if cryptographic protocols and algorithms are used).
+
+**Met**
+all crypto uses openssl
+
+> If the software produced by the project is an application or library, and its primary purpose is not to implement cryptography, then it SHOULD only call on software specifically designed to implement cryptographic functions; it SHOULD NOT re-implement its own.
+
+**Met**
+all crypto uses openssl
```

Comment:
@tniessen is this still true?

Reply:
For the most part. It's up to interpretation. We currently use the https://github.com/quictls/openssl fork for all low-level cryptographic operations.

That being said, we do implement a few oddities of the Web Crypto API ourselves, but still based on OpenSSL primitives. I guess the Node.js crypto module could also be seen as "software specifically designed to implement cryptographic functions" as the document says, so it really is up for interpretation.

Reply:
Maybe we can improve the explanation, like `Met. All crypto operations use directly OpenSSL or build functionalities on top of OpenSSL primitives`